coherence-distributed-configuration/src/main/java/io/micronaut/coherence/discovery/CoherenceConfigurationClient.java

```diff
@@ -0,0 +1,182 @@
+            Map<String, Object> props = configCache.entrySet(filter)
+                    .stream()
+                    .collect(Collectors.toMap((Converter<Map.Entry<String, Object>, String>)
+                                    cacheEntry -> cacheEntry.getKey().substring(propertySourcePath.length()),
+                            Map.Entry::getValue));
+            Flowable<PropertySource> propertySourceFlowable = Flowable.just(PropertySource.of(propertySource, props, priority));
+            propertySources.add(propertySourceFlowable);
+        }
+        return Flowable.merge(propertySources);
+    }
+
+    /**
+     * Returns NamedCache instance.
+     *
+     * @param cacheName                    cache name
+     * @param coherenceClientConfiguration configuration
+     * @return cache
+     */
+    protected NamedCache<String, Object> getCache(String cacheName, CoherenceClientConfiguration coherenceClientConfiguration) {
+        Session session = buildSession(coherenceClientConfiguration);
+        return session.getCache(cacheName);
+    }
+
+    /**
+     * Builds Coherence session.
+     *
+     * @param coherenceClientConfiguration configuration
+     * @return Coherence session
+     */
+    protected Session buildSession(CoherenceClientConfiguration coherenceClientConfiguration) {
+        Channel channel = buildChannel(coherenceClientConfiguration);
+
+        GrpcSessionConfiguration.Builder builder = GrpcSessionConfiguration.builder(channel);
+        GrpcSessionConfiguration grpcSessionConfiguration = builder.build();
+
+        Optional<Session> optional = Session.create(grpcSessionConfiguration);
+        return optional.get();
+    }
+
+    /**
+     * Builds gRPC channel.
+     *
+     * @param coherenceClientConfiguration configuration
+     * @return gRPC channel
+     */
+    protected Channel buildChannel(CoherenceClientConfiguration coherenceClientConfiguration) {
+        String host = coherenceClientConfiguration.getHost();
+        int port = coherenceClientConfiguration.getPort();
+
+        ManagedChannelBuilder<?> chanelBuilder = ManagedChannelBuilder.forAddress(host, port);
+        if (!coherenceClientConfiguration.isEnableTls()) {
+            chanelBuilder.usePlaintext();
+        }
+        return chanelBuilder.build();
+    }
+
+    /**
+     * Builds the keys used to get properties.
+     *
+     * @param applicationConfiguration The application configuration
+     * @param environment              environment
+     * @return map of precedence values and folders
+     */
+    protected Map<Integer, String> buildKeys(ApplicationConfiguration applicationConfiguration, Environment environment) {
+        Optional<String> configuredApplicationName = applicationConfiguration.getName();
+        String applicationName = configuredApplicationName.orElse(null);
+        Set<String> environmentNames = environment.getActiveNames();
+
+        Map<Integer, String> vaultKeys = new HashMap<>();
```

Comment:
Why `vaultKeys`? I'm assuming this is copy&paste leftover from Vault implementation?

`configKeys` would make a lot more sense.

Reply:
Fixed.